osquery/core/init.cpp

```diff
@@ -570,11 +573,25 @@ void Initializer::start() const {
     DatabasePlugin::setAllowOpen(true);
     // A daemon must always have R/W access to the database.
     DatabasePlugin::setRequireWrite(tool_ == ToolType::DAEMON);
-    if (!DatabasePlugin::initPlugin()) {
-      LOG(ERROR) << RLOG(1629) << binary_
-                 << " initialize failed: Could not initialize database";
-      auto retcode = (isWorker()) ? EXIT_CATASTROPHIC : EXIT_FAILURE;
-      requestShutdown(retcode);
+
+    for (size_t i = 1; i <= kDatabaseMaxRetryCount; i++) {
+      auto ok = DatabasePlugin::initPlugin();
+      if (ok) {
+        break;
+      }
+
+      if (i == kDatabaseMaxRetryCount) {
+        LOG(ERROR) << RLOG(1629) << binary_
+                   << " initialize failed: Could not initialize database";
+        auto retcode = (isWorker()) ? EXIT_CATASTROPHIC : EXIT_FAILURE;
+        requestShutdown(retcode);
+      }
+
+      LOG(WARNING) << RLOG(1629) << binary_
```

Comment:
I'd recommend removing these. If the goal is to wait a few moments for the database to be ready, you can instead check a lot faster, every 100/200ms for example, up until a max of 5 seconds have passed.